File: webapp/frontend/app/src/app/model/skills/skill.ts
export interface SkillMetadata {
	name: string;
	description: string;
	owned: boolean;
}

export interface SkillFile {
	relativePath: string;
	content: string;
}

export interface Skill {
	name: string;
	metadata: SkillMetadata;
	files: SkillFile[];
}

File: webapp/backend/api/src/main/java/tom/api/skill/Skill.java
package tom.api.skill;

import java.util.List;

public record Skill(String name, SkillMetadata metadata, List<SkillFile> files) {
	SkillFile mainFile() {
		return files.stream().filter(f -> f.relativePath().equals("SKILL.md")).findFirst()
				.orElseThrow(() -> new IllegalStateException("No SKILL.md in skill: " + name));
	}
}
